src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -265,6 +265,36 @@ public String createCitationRIS(DatasetVersion version, FileMetadata fileMetadat
         return retString;
     }
 
+    public String createCitationBibtex(DatasetVersion version) {
+        return createCitationBibtex(version, null);
+    }
+
+    public String createCitationBibtex(DatasetVersion version, FileMetadata fileMetadata) {
```

Comment:
@bmckinney I mentioned a while ago that it would be great to have unit tests for this `createCitationBibtex` method but it hasn't been until today at http://irclog.iq.harvard.edu/dataverse/2016-03-22#i_33154 that I got around to explaining my suggestion. As we discussed, it boils down to using `DatasetVersionDTO` and `FileMetadataDTO` objects rather than the non-DTO "entity" equivalents you're using now. To see some DTO objects in action see https://github.com/IQSS/dataverse/blob/v4.3/src/main/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtil.java and the unit tests that exercise them at https://github.com/IQSS/dataverse/blob/v4.3/src/test/java/edu/harvard/iq/dataverse/export/ddi/DdiExportUtilTest.java

The advantage of using these DTO objects is that you can run the unit tests without having PostgreSQL running.

@scolapasta just asked @kcondon to take a look at this pull request (#3020) so you may want to ping @scolapasta on if he'd like you to add unit tests.